main.sh

```diff
@@ -153,6 +153,9 @@ fi
 for tool in "${tools[@]}"; do
     if [[ "${tool}" == *"@"* ]]; then
         version="${tool#*@}"
+        if [[ "${version}" != "latest" ]] && ! [[ "${version}" =~ ^[0-9]*\.[0-9]*(\.[0-9]*)?$ ]]; then
```

Comment:
Well, the semver version is actually a bit more complicated because it allows prerelease and metadata.
FYI, if we accept only the exact valid semver version, probably then:

```sh
if [[ ! "${version}" =~ ^[1-9][0-9]*(\.(0|[1-9][0-9]*)(\.(0|[1-9][0-9]*)(-[0-9A-Za-z\.-]+)?(\+[0-9A-Za-z\.-]+)?)?)?$|^0\.[1-9][0-9]*(\.(0|[1-9][0-9]*)(-[0-9A-Za-z\.-]+)?(\+[0-9A-Za-z\.-]+)?)?$|^0\.0\.(0|[1-9][0-9]*)(-[0-9A-Za-z\.-]+)?(\+[0-9A-Za-z\.-]+)?$|^latest$ ]]; then
```

However, we don't need something so exacting, so the following should probably be fine:


```suggestion
        if [[ ! "${version}" =~ ^[1-9][0-9]*$|^[0-9]+\.[1-9][0-9]*$|^[0-9]+\.[0-9]+\.[0-9]+(-[0-9A-Za-z\.-]+)?(\+[0-9A-Za-z\.-]+)?$|^latest$ ]]; then
```